src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1270,19 +1270,54 @@
   (let [f (first expr)]
     (and (str f) (= f "extract"))))
 
+(defn get-clause
+  [clause clauses]
+  (when-let [candidates (seq (filter #(= (first %) clause) clauses))]
+    (cond
+      (> (count candidates) 1)
+      (throw (IllegalArgumentException.
+               (format "Multiple '%s' clauses are not permitted" clause)))
+
+      (not (second (first candidates)))
+      (throw (IllegalArgumentException.
+               (format "Received '%s' clause without an argument" clause))))
+
+      :else (second (first candidates))))
+
+(defn process-order-by
+  [clauses]
+  (when clauses
+    (for [clause clauses]
+      (if (vector? clause) (mapv keyword clause) (keyword clause)))))
+
+(defn create-paging-map
```

Comment:
I'd like to see a hint in here somewhere, with a schema or docstring or something, of what the data looks like coming into and going out of this function.